Codegen reimplementation for GraphQL unions (#666)
graphql-rust/juniper: juniper/CHANGELOG.md

- Derive macro `GraphQLEnum` supports custom context (see [#621](https://github.com/graphql-rust/juniper/pull/621))

> - Reworked `#[derive(GraphQLUnion)]` macro ([#666]):
>     - applicable to enums and structs;
>     - supports custom resolvers;
>     - supports generics;

Reviewer: ```suggestion
- Reworked `#[derive(GraphQLUnion)]` macro ([#666]):
    - Applicable to enums and structs.
    - Supports custom resolvers.
    - Supports generics.
```

    - supports multiple `#[graphql]` attributes.

> - New `#[graphql_union]` macro ([#666]):
>     - applicable to traits;
>     - supports custom resolvers;
>     - supports generics;
>     - supports multiple `#[graphql_union]` attributes.

Reviewer: ```suggestion
- New `#[graphql_union]` macro ([#666]):
    - applicable to traits;
    - Supports custom resolvers.
    - Supports generics.
    - Supports multiple `#[graphql_union]` attributes.
```

- Remove deprecated `ScalarValue` custom derive (renamed to GraphQLScalarValue)

> - `graphql_union!` macro removed, replaced by `#[graphql_union]` proc macro and custom resolvers of `#[derive(GraphQLUnion)]` macro.

Reviewer: ```suggestion
- `graphql_union!` macro removed, replaced by `#[graphql_union]` proc macro and custom resolvers for the `#[derive(GraphQLUnion)]` macro.
```

> - `#[derive(GraphQLUnion)]` macro doesn't generate `From` impls for enum variants anymore, consider [`derive_more`](https//docs.rs/derive_more)) crate to do that ([#666]).

Reviewer: ```suggestion
- The `#[derive(GraphQLUnion)]` macro doesn't generate `From` impls for enum variants anymore. Consider using the [`derive_more`](https//docs.rs/derive_more)) crate directly ([#666]).
```
